service/audit-service/src/main/java/uk/gov/gchq/palisade/audit/service/request/AuditRequestWithException.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.audit.service.request;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.ToStringBuilder;
+import uk.gov.gchq.palisade.User;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This is one of the objects that is passed to the {@link uk.gov.gchq.palisade.audit.service.AuditService}
+ * to be able to store an audit record. This class extends {@link AuditRequest} This class
+ * is used for the indication to the Audit logs that an exception has been received.
+ */
+public class AuditRequestWithException extends AuditRequest {
+    private Throwable exception;
+    private User user;
+
+    public AuditRequestWithException() {
+    }
+
+    @Override
+    public String constructAuditLog() {
+        final String msg = "AuditRequestWithException: " + getUser().getUserId().getId()
```

Comment:
There is an instance when you would call this without having set the User, which would result in you getting a null pointer exception. I would expect that any AuditRequestWithException would just need to have the request ID (an id that uniquely references this request for data so the full flow of the request can be tracked, which having a quick look might be missing) and the exception.